near-sdk-macros/build.rs

```diff
@@ -0,0 +1,10 @@
+fn main() {
+    if cfg!(feature = "__abi-embed") {
+        if option_env!("CARGO_NEAR_ABI_PATH").is_some() {
+            println!("cargo:rustc-cfg=feature=\"__abi-embed-checked\"");
```

Comment:
curious -- I wonder if it is possible to manually pass the feature flag to the compiler from outside of the build script? This way we would be able to remove the pseudo-private flags altogether since the overhead doesn't matter for cargo-near?

Reply:
You mean through `RUSTFLAGS`? That applies to all `rustc` instances cargo spawns throughout its execution. Can't guarantee anything else (deps) doesn't change its behavior with that config. But through `cargo`, we constrain the activation to the crate in particular. And then only conditionally activate `__abi-embed-checked`, also specifically for the crate.

An alternative I experimented with early on, was to use [`cargo-near` as `RUST_WRAPPER`](https://github.com/near/cargo-near/blob/bcf9ae57ec3302a979f450b200113f74c61a97c7/cargo-near/src/contract.rs#L81), so when generating ABI, it embeds a unique identifier for the sdk in it's chunks. And when embedding the ABI, it selectively [injects the embed flag](https://github.com/near/cargo-near/blob/bcf9ae57ec3302a979f450b200113f74c61a97c7/cargo-near/src/rustc_wrapper.rs#L45) specifically for the SDK.

But that was cumbersome, so I stripped it.

Reply:
> Can't guarantee anything else (deps) maintains its behavior with that flag activated

What would be the concern there, that there would be a dependency used in a contract being built that for some reason uses a `__abi-embed` feature flag?

I think this solution is good enough for now, but maybe we could have at least a tracking issue to see if we can get rid of these pseudo-private feature flags

Reply:
> What would be the concern there, that there would be a dependency used in a contract being built that for some reason uses a __abi-embed feature flag?

Yup, that's the primary concern. Arguably, a good approach would be to localize effects instead of globalizing.